From: Justus Winter <justus@g10code.com>
Date: Tue, 28 Mar 2017 12:10:28 +0200
Subject: g10: Fix memory leak.

* g10/decrypt-data.c (decrypt_data): Free 'filename'.

Signed-off-by: Justus Winter <justus@g10code.com>
(cherry picked from commit 6d3edfd972c1114f43f6b35773dc25e0256f48f4)
---
 g10/decrypt-data.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/g10/decrypt-data.c b/g10/decrypt-data.c
index 585b150..f5843d6 100644
--- a/g10/decrypt-data.c
+++ b/g10/decrypt-data.c
@@ -222,7 +222,7 @@ decrypt_data (ctrl_t ctrl, void *procctx, PKT_encrypted *ed, DEK *dek)
 
   if (opt.unwrap_encryption)
     {
-      char *filename;
+      char *filename = NULL;
       estream_t fp;
       rc = get_output_file ("", 0, ed->buf, &filename, &fp);
       if (! rc)
@@ -248,6 +248,7 @@ decrypt_data (ctrl_t ctrl, void *procctx, PKT_encrypted *ed, DEK *dek)
           if (afx)
             release_armor_context (afx);
         }
+      xfree (filename);
     }
   else
     proc_packets (ctrl, procctx, ed->buf );
